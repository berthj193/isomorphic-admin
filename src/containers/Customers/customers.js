import React, { Component } from 'react';
import PropTypes from 'prop-types';
import columns from './columns';
import customersQueries from '../../graphql/customers';
import ItemList from '../../components/list/item-list';
import withOrdersManager from '../../components/apolloProviders/withOrdersManager';

export class Customers extends Component {
  static propTypes = {
    customers: PropTypes.arrayOf(
      PropTypes.shape({
        auth0_id: PropTypes.string,
        given_name: PropTypes.string,
        family_name: PropTypes.string,
        email: PropTypes.string,
      }),
    ),
    error: PropTypes.string,
  };

  static defaultProps = {
    customers: [],
  };

  dataSource() {
    return this.props.customers
      .map(customer => ({
        key: customer.auth0_id,
        id: customer.auth0_id,
        name: `${customer.given_name} ${customer.family_name}`,
        email: customer.email,
      }));
  }

  render() {
    return (
      <ItemList
        data={this.dataSource()}
        error={this.props.error}
        title="Customers"
        columns={columns}
      />
    );
  }
}

export default withOrdersManager(Customers, {
  query: customersQueries.getList(),
});
