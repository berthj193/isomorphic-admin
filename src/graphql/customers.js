import gql from 'graphql-tag';

export const getListQuery = () => `
query {
  customers {
    given_name
    family_name
    auth0_id
    email

    addresses {
      line_one
      postcode
    }

    payment_sources {
      id
      last4
      brand
      payment_method
      created_at
      updated_at
    }

    created_at
    updated_at
  }
}
`;

export default {
  getList: () => gql(getListQuery()),
};
