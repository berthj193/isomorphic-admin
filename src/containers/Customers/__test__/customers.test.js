import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { Customers } from '../customers';

describe('<Customers />', () => {
  it('should render correctly', () => {
    expect(toJson(shallow(<Customers />))).toMatchSnapshot();
  });

  it('should render error correctly', () => {
    expect(toJson(shallow(<Customers error="Woopsie, error happened" />))).toMatchSnapshot();
  });

  it('should map customers to proper object', () => {
    const mockData = [
      {
        auth0_id: 'holonet|d3str0y-d3ath-st4r',
        given_name: 'Leia',
        family_name: 'Organa',
        email: 'lorgana@rebellion.ga',
      },
      {
        auth0_id: 'imperialnet|1-am-th3-s3n4t3',
        given_name: 'Sheev',
        family_name: 'Palpatine',
        email: 'palpi@coruscant.co',
      },
    ];
    const instance = shallow(<Customers customers={mockData} />).instance();
    expect(instance.dataSource()).toMatchSnapshot();
  });
});
