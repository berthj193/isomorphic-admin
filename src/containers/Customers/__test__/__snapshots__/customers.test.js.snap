// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Customers /> should map customers to proper object 1`] = `
Array [
  Object {
    "email": "lorgana@rebellion.ga",
    "id": "holonet|d3str0y-d3ath-st4r",
    "key": "holonet|d3str0y-d3ath-st4r",
    "name": "Leia Organa",
  },
  Object {
    "email": "palpi@coruscant.co",
    "id": "imperialnet|1-am-th3-s3n4t3",
    "key": "imperialnet|1-am-th3-s3n4t3",
    "name": "Sheev Palpatine",
  },
]
`;

exports[`<Customers /> should render correctly 1`] = `
<withRouter(ItemList)
  columns={
    Array [
      Object {
        "dataIndex": "id",
        "key": "id",
        "title": "ID",
      },
      Object {
        "dataIndex": "name",
        "key": "name",
        "title": "Name",
      },
      Object {
        "dataIndex": "email",
        "key": "email",
        "title": "Email",
      },
      Object {
        "dataIndex": "orders",
        "key": "orders",
        "title": "Orders",
      },
    ]
  }
  data={Array []}
  title="Customers"
/>
`;

exports[`<Customers /> should render error correctly 1`] = `
<withRouter(ItemList)
  columns={
    Array [
      Object {
        "dataIndex": "id",
        "key": "id",
        "title": "ID",
      },
      Object {
        "dataIndex": "name",
        "key": "name",
        "title": "Name",
      },
      Object {
        "dataIndex": "email",
        "key": "email",
        "title": "Email",
      },
      Object {
        "dataIndex": "orders",
        "key": "orders",
        "title": "Orders",
      },
    ]
  }
  data={Array []}
  error="Woopsie, error happened"
  title="Customers"
/>
`;
